Server/Classes/Handlers/VehicleSpawner.sqx

```diff
@@ -125,12 +125,69 @@ namespace Intrusion.Server
 		    clearBackpackCargoGlobal _vehicleObj;
 		    clearItemCargoGlobal _vehicleObj;
 		    
-		    {
-            	_vehicleObj addBackpackCargoGlobal [_x, 1];
-		    } foreach _vehicleClassInfo.CargoWeapons as String;
-		    
         	_intrusionVehicle = [_vehicleObj, _side, _professionType, _group, _cost] new IntrusionVehicle;
         	_vehicleObj setVariable ["IntrusionVehicle", _intrusionVehicle, true];
+        	
+	    	{
+	    		private ["_attachObj" as Object, "_attachObjInfo" as VehicleAttachmentInfo];
+	    		_attachObjInfo = _x;
+	    		
+	    		_attachObj = _attachObjInfo.ClassName createVehicle (getPos _vehicleObj);
+				_attachObj attachTo [_vehicleObj, _attachObjInfo.AttachOffset];
+				clearWeaponCargoGlobal _attachObj;
+				clearItemCargoGlobal _attachObj;
+				clearMagazineCargoGlobal _attachObj;
+				
+				// Fill the weaponbox again
+				{
+	            	if (_x.ClassName isKindOf ["Bag_Base", configFile>>"CfgVehicles"]) then {
+			    		_vehicleObj addBackpackCargoGlobal [_x.ClassName, _x.Quantity];
+			    	} else {
+	    	        	_vehicleObj addItemCargoGlobal [_x.ClassName, _x.Quantity];
+	    	        };
+			    } foreach _vehicleClassInfo.CargoItems as VehicleCargoInfo;
+			    
+			    // When this is a part of the vehicle, add the IntrusionVehicle object.
+				// This will lock the crate to right profession.
+			    _attachObj setVariable ["IntrusionVehicle", _intrusionVehicle, true];
+			    
+			    // When vehicle explodes, destroy ammo box
+				_vehicleObj addEventHandler ["Killed", {
```

Comment:
Is this tested with a player in the vehicle? When a player is in a vehicle, the vehicle is local to the player's machine. Is this event handler working as expected then?